File: cs1210/2019summer/inclass/wk06_2019-07-18_thu/factorial.py
def factorial(n):
    # Check for base case
    if n <= 1:
        return 1 # this handles 0, 1, and all negative numbers
    
    return n * factorial(n-1)

def main():
    num = 12
    fact = factorial(num)
    print(f'{num}! = {fact}')
    
main()

File: csci294/2020Spring/inclass/wk04_2020-02-03_mon/Factorial.cpp
/* Factorial.cpp */
// A recursive function to computer n!

#include <iostream>
using namespace std;

int fact(int);

int main() {
    cout << fact(3) << endl;
}

// Computer n! using recursion
int fact(int n){
    // check for base case
    if(n <= 1){
        return 1;
    }
    
    return n * fact(n - 1);
}